(feat) Integrate month-day-preview, integrate month with backend

joshuaNewman10 opened the pull request.

Commit:
(feat) Integrate month-day-preview, integrate month with backend

(feat) Add additional components to month view to preview day and analyze scheduling

(feat) Create styling for month-day-previw

(feat) Add classes to month components to work with month design

(feat)Add db config folder to gitignore

(feat) Implement month data requests from prev/next month buttons

(refactor) Change name of event

(refactor) Change format of mock month data

(feat) Add internal func to put data in expected order

(feat) Make get request for month data

(feat) Correctly load month data from server

(bugfix) Fix bug in month store

(styling) Add styling for day preview in month component

(feat) Add component for each event in month component

(feat) Add preview of day data within moneht component

(feat) Add dispatcher event to listen for SHOW_EVENT_PREVIEW

(feat) add view action to show event preview in month component

(feat) Add new constant for showing event preview in month component

(bugfix) Fix data data request in api-utils

(feat) Dynamically add events for given day in month component

(refactor) Correctly accept and render month data

(bugfix) Make month's day preview make a API request for the correct date

(feat) Fully implement month component

(bugfix) Fix dates that weren't correctly translating their format

(bugfix) Fix spelling mistake on month

(bugfix) Fix conflicts after mege

(refactor) Remove use of temp month data

kyleaowen commented on client/js/monthData.js:
```diff
@@ -1,151 +1,48 @@
 /*Temporary Mock Month Data */
```
are you using this data somewhere?

joshuaNewman10 replied:
So if our server was down I wanted to render an empty calendar. But just talked to Ryan and we agreed that it makes more sense to render nothing. That way we won't mislead the users about there not being any events for a given day

its been added to gitignore

kyleaowen replied:
is it being used as defaults somewhere? I'd imagine that if our server was down, we'd want to display an error message and ask users to check back later or something

kyleaowen commented on client/js/stores/day-store.js:
```diff
@@ -53,13 +54,12 @@ var DayStore = assign({}, EventEmitter.prototype, {
       });
       _dayData.date = payload.date;
       DayStore.emitChange();
-      break;
```
why did you remove this break?

kyleaowen commented on client/js/stores/month-store.js:
```diff
@@ -14,6 +14,50 @@ var _setMonthData = function (data) {
+  currentDate = moment(data.startDate, 'YYYY-MM-DD');
+  for(var j=1; j<=numDays; j++) {
+    dataKey = currentDate.format('YYYYMMDD');
+    weekNum = Math.ceil(j / 7);
```
the variable weekNum is missing a declaration.

rjatkinson2 commented on client/css/eventfull.css:
```diff
@@ -242,6 +242,11 @@ input[type=submit]{
+  margin-right: 0px;
+}
+
+.day-wrapper {
```
What's going on with this class?

joshuaNewman10 replied:
Good catch! Removed!

rjatkinson2 commented on client/js/actions/server-action-creator.js:
```diff
@@ -21,9 +21,9 @@ var ServerActionCreator = {
     });
   },
-  fetchedMonthData: function (monthData) {
+  receiveMonthData: function (monthData) {
```
Glad we are normalizing our verbiage.

rjatkinson2 commented on client/js/components/month/month.js:
```diff
@@ -1,41 +1,43 @@
-      'week2': [{date: '', numEvents: 0, status: ''}],
-      'week3': [{date: '', numEvents: 0, status: ''}],
-      'week4': [{date: '', numEvents: 0, status: ''}]
+      startDate : '2015-05-10',
```
What's the meaning of this start date? Should be able to just use the current day.

rjatkinson2 commented on client/js/components/month/month.js:
```diff
@@ -1,41 +1,43 @@
     };
   },
   componentWillMount: function () {
+    var today = moment().startOf('month');
```
I don't think the variable name today make sense if it is assigned to the first day of the month?

rjatkinson2 commented on client/js/components/month/month-prev-button.js:
```diff
@@ -1,19 +1,19 @@
 
-module.exports = MonthPrevButton;
\ No newline at end of file
+module.exports = MonthPrevButton;
```
These two files (MonthNext and MonthPrev) are exactly the same except for the className and the date method. You could break this down to a single component and pass the className and method in as props.

joshuaNewman10 replied:
Cool, I thought it made more sense to break them out into two seperate components since they are pretty short and we may want to differentiate them in some fundamental way. But im happy to make them into one reusable component since that is what we have done for week and day

kyleaowen replied:
If you want to make an arrow header abstraction, allow to take the parameters "direction" and "timeSpan". direction can be "left" or "right", timeSpan can be "day" "month" or "week"

kyleaowen replied:
that may be something we could refactor later. for now, it may be easiest to just do something similar to the day-header, where the logic is all in one place

rjatkinson2 commented on client/js/components/month/month-header.js:
```diff
@@ -8,12 +8,12 @@ var MonthHeader = React.createClass({
-        <div className="arrow arrow-left"></div>
-        <h2>May 2015</h2>
-        <div className="arrow arrow-right"></div>
+        <PrevMonth className="arrow arrow-left" month={this.props.month} year={this.props.year} />
```
classNames should only exist on the render method of the component. if you want them to be dynamic, you could pass them in as props.

rjatkinson2 commented on client/js/components/month/month-day-preview.js:
```diff
@@ -0,0 +1,45 @@
+      });
+    }
+    return (
+      <div className = "col-md-3" onClick={ this.toggleDetails }>
```
I don't see toggleDetails in this component?

rjatkinson2 commented on client/js/components/month/month-day-event.js:
```diff
@@ -0,0 +1,23 @@
+    var date = moment(this.props.eventData.date, 'YYYY-MM-DD');
+    AppViewActions.getDayData(date);
+  },
+  render: function () {
```
This is a bit repetitive. Maybe something like:

``` javascript
var color = this.props.health === 10 ? 'green' : this.props.health < 4 ? 'red' : 'blue';
return <div onClick={ this.handleClick } className={ 'events-circle' + color } ></div>;
```

joshuaNewman10 replied:
Good call, much simpler!

rjatkinson2 commented on client/js/components/month/month-day-card.js:
```diff
@@ -1,31 +1,25 @@
   render:function () {
-    var currentDate = this.props.dayData.date;
-    var dayPath = '/day/' + currentDate;
+    var weekNumber = Number(this.props.dayData.weekNumber);
```
I don't see this used again.

joshuaNewman10 replied:
Good catch, it was from an old version where I needed to pass in the week and day and do a calculation based on that

rjatkinson2 commented on client/js/components/month/month-day-card.js:
```diff
@@ -1,31 +1,25 @@
-    var currentDate = this.props.dayData.date;
-    var dayPath = '/day/' + currentDate;
+    var weekNumber = Number(this.props.dayData.weekNumber);
+    var dayNumber = parseInt(moment(this.props.dayData.dayNumber, 'YYYYMMDD').format('DD'), 10);
```
Why do you need to parseInt here?

joshuaNewman10 replied:
It was from when moment was causing problems and not converting to the correct day as expected, it will be removed in my corrections

kyleaowen commented on client/js/components/month/month-day-preview.js:
```diff
@@ -0,0 +1,45 @@
+    date = date.format('MMMM Do');
+    var data = [];
+    if ( this.state.gigs !== null ) {
+      data = _.map(this.state.gigs, function (gig, key) {
```
what is the purpose of this map? it seems like you could pass this.state.gigs directly into the WeekDayBin and it would work similarly

kyleaowen commented:
I noticed that ViewActionCreator is sometimes being called AppViewActions. I think we should stick with ViewActionCreator, which is present across the rest of our codebase, thoughts?

joshuaNewman10 committed:
(refactor) Implement PR changes

(bugfix) Remove monthData from being followed

(bugfix) Bring back break statement

(bugfix) Fix changes from PR comments

kyleaowen commented on client/js/components/month/month-nav-arrow.js:
```diff
@@ -0,0 +1,20 @@
+/** @jsx React.DOM */
+var React = require('react');
+var AppViewActions = require('../../actions/view-action-creator');
```
ViewActionCreator?

rjatkinson2 commented:
Thanks for rocking the changes!
